Documentation: How to contribute (#294)
SEKOIA-IO/intake-formats: doc/format.md

- a `manifest.yml` document.
  This document must contain an UUID, that identify the module, the name of the module, a short and url-friendly name of the module and a short description that describes the module.
 
  For example, you can inspire from the [AWS Module manifest](../AWS/_meta/manifest.yml).

### Formats

A format is associated to a software technology. In general, a format correspond to a vendor product.

Each format share the same tree structure:
- `_meta`, a meta-information directory
- `ingest/parser.yml`, the parser of the format
- a set of testings files

#### Format `_meta` directory

Like the module one, this directory holds meta-information about the format.
It consists of 4 files:

- A logo that identify the product. This logo should help to identify the product the format represent.

  It must be a PNG image, with transparent background, named `logo.png` and must be lighter than 50Ko.

- a `manifest.yml` document.
  This document must contain an UUID, that identify the format, the name of the format, a short and url-friendly name, a short description that describes the software technology and a list of [data-sources](datasources.md).
  

>   For example, you can inspire from [Trend Micro Deep Security manifest](../Trend%20Micro/deep-security/_meta/manifest.yml)

Reviewer: ```suggestion
  Here is an example from [Trend Micro Deep Security manifest](../Trend%20Micro/deep-security/_meta/manifest.yml)
```

> - a [taxonomy](taxonomy.md) file, that describe fields used in the parser.

Reviewer: ```suggestion
- A [taxonomy](taxonomy.md) file that describes fields used in the parser.
```

  For inspiration, see [Windows taxonomy](../Windows/windows/_meta/fields.yml).
  

> - some [smart-descriptions](smart-descriptions.md) (e.g: [Windows smart-descriptions](../Windows/windows/_meta/smart-descriptions.json))

Reviewer: ```suggestion
- Some [smart-descriptions](smart-descriptions.md) (e.g: [Windows smart-descriptions](../Windows/windows/_meta/smart-descriptions.json))
```

#### Parser

The parser transforms raw event into [structured event](structured_event.md).

It's a vocabulary scheduling how information will be extracted from the raw event.

> it consists of a pipeline, setting up the sequence of data extraction, and stages, defining the construction of the structured event.

Reviewer: ```suggestion
It consists of a pipeline setting up the sequence of data extraction as well as stages defining the construction of the structured event.
```
